### DISCUSSION TITLE
WinGetUI lists Python updates, even though Miniconda is up2date

### DISCUSSION THREAD

**Discussion by bjspi** [Devolutions/UniGetUI#2100] [Q&A] [answered]
Hi community,

I'm wondering which information is taken into account by WinGetUI about installed Python packages? 
Because it lists quite a lot of pip and python packages to be installed, which seems to be meant for my miniconda distribution.

![image](https://github.com/marticliment/WingetUI/assets/43368639/d6a0fffd-4ca7-42c4-877e-bf4c6c5e9959)

My question is:
Why does WinGetUI think these packages are outdated!?

Miniconda shows, all are up2date, using "conda update --all":
![image](https://github.com/marticliment/WingetUI/assets/43368639/b2c4a11c-153c-4b4a-be52-1b5534f8f494)

So why is WinGetUI still suggesting to update them?
Not sure if that's safe to do - obviously doesnt seem to "fit" to my Miniconda Distro?

Thank you a lot

**Comment by marticliment** [comment, marked as answer]
Hi @bjspi, WingetUI uses pip to fetch Python updates. 

You may want to find out why does pip report updates (perhaps the conda environment is isolated from the default python-pip one?), or you may want to disable pip from WingetUI settings.

**Reply by bjspi** [reply]
@marticliment 
Thank you for your response. Very interesting! Maybe PIP and CONDA are not fully synchronized or it's by design to not update to a new major version on the go when calling "conda update --all".

I dig deeper today and found that manually calling pip will indeed show an update, so I did it:
![image](https://github.com/marticliment/WingetUI/assets/43368639/d1e861ec-daa0-4102-b118-8532faa547da)

Even though PIP itself shows only a small lits of possibly outdated packages:
![image](https://github.com/marticliment/WingetUI/assets/43368639/e8586cc5-80d0-49de-bb3c-7c272858af15)

I als installed "pip-check" as suggested by a few forums online, found it matches the list of the above pip outdated list:
![image](https://github.com/marticliment/WingetUI/assets/43368639/4f05dc77-92a8-4fe5-a927-352459116078)

It's all interesting, but not of major concern - I was just surprised to see it and thought of reporting it back.
But yeah, in the end, I'll manage the Python Env by Miniconda anyhow hich takes care of versions and dependencies. 

**EDIT** Argh, totally forgot about my base environment.
So yeah, this is the solution, everything is fine than
![image](https://github.com/marticliment/WingetUI/assets/43368639/5aa5dc40-1795-40a1-91fb-c3b7833b5bac)
